app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/view/DisplayDatasetActivity.kt

```diff
@@ -0,0 +1,91 @@
+package com.github.HumanLearning2021.HumanLearningApp.view
+
+import android.app.Activity
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import android.os.Parcelable
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.BaseAdapter
+import android.widget.GridView
+import android.widget.ImageView
+import android.widget.TextView
+import androidx.appcompat.app.AppCompatActivity
+import androidx.lifecycle.lifecycleScope
+import com.github.HumanLearning2021.HumanLearningApp.R
+import com.github.HumanLearning2021.HumanLearningApp.model.CategorizedPicture
+import com.github.HumanLearning2021.HumanLearningApp.model.DummyCategory
+import com.github.HumanLearning2021.HumanLearningApp.presenter.DummyUIPresenter
+import kotlinx.coroutines.launch
+import java.io.Serializable
+
+
+class DisplayDatasetActivity : AppCompatActivity() {
+
+    val fork = DummyCategory("fork")
+    val knife = DummyCategory("knife")
+    val spoon = DummyCategory("spoon")
+
+    val dummyPresenter = DummyUIPresenter()
+
+    val datasetImagesList = ArrayList<CategorizedPicture>()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_display_dataset)
+        lifecycleScope.launch {
+            datasetImagesList.add(dummyPresenter.getPicture(fork.name)!!)
+            datasetImagesList.add(dummyPresenter.getPicture(knife.name)!!)
+            datasetImagesList.add(dummyPresenter.getPicture(spoon.name)!!)
+        }
+
+        val displayDatasetAdapter = DisplayDatasetAdapter(datasetImagesList, this)
+
+        findViewById<GridView>(R.id.display_dataset_imagesGridView).adapter = displayDatasetAdapter
+
+        findViewById<GridView>(R.id.display_dataset_imagesGridView).setOnItemClickListener { adapterView, view, i, l ->
+            val intent = Intent(this, DisplayImageActivity::class.java)
+            intent.putExtra("display_image_image", (datasetImagesList[i]) as Serializable)
+            startActivity(intent)
+        }
+
+
+    }
+
+    class DisplayDatasetAdapter(
+        private var images: ArrayList<CategorizedPicture>,
+        private var context: Context
```

Comment:
```suggestion
        private val context: Activity
```
that should get rid of the downcast. In fact, you can use an [inner class](https://kotlinlang.org/docs/nested-classes.html#inner-classes) here.